src/main/java/uk/gov/companieshouse/transactions/web/interceptor/LoggingInterceptor.java

```diff
@@ -1,31 +1,32 @@
 package uk.gov.companieshouse.transactions.web.interceptor;
 
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.lang.NonNull;
 import org.springframework.lang.Nullable;
 import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
 import org.springframework.web.servlet.ModelAndView;
-import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 import uk.gov.companieshouse.logging.util.RequestLogger;
 import uk.gov.companieshouse.transactions.web.TransactionsWebApplication;
 
 @Component
-public class LoggingInterceptor extends HandlerInterceptorAdapter implements RequestLogger {
+public class LoggingInterceptor implements HandlerInterceptor, RequestLogger {
 
     private static final Logger LOGGER =
             LoggerFactory.getLogger(TransactionsWebApplication.APPLICATION_NAME_SPACE);
 
     @Override
-    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+    public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull Object handler) {
 
```

Comment:
question: why you are marking request response objects as nonnull as they wont be null - as they are managed by servlet container (You dont need to change it but curious to know)

Reply:
it was actually the IDE code analysis that suggested this with the message: 
"Not annotated parameter overrides @NonNullApi parameter"

And the suggested resolution was to add the @NonNull annotation to each parameter! :)